src/Compilers/Core/MSBuildTaskTests/TestUtilities/IntegrationTestBase.cs

```diff
@@ -119,72 +120,151 @@ protected static ProcessResult RunCommandLineCompiler(
+        if (!useAppHost)
+        {
+            _output.WriteLine($"Apphost: {originalAppHost}");
+            File.Move(originalAppHost, backupAppHost);
+        }
+
+        ProcessResult? result;
+
+        try
+        {
+            result = RunMsbuild(
+                "/v:n /m /nr:false /t:Build /restore Test.csproj" +
+                    (overrideToolExe ? $" /p:CscToolExe=csc{PlatformInformation.ExeExtension}" : ""),
+                _tempDirectory,
+                new Dictionary<string, string>
+                {
+                    { "File.cs", """
+                        class Program { static void Main() { System.Console.WriteLine("Hello from file"); } }
+                        """ },
+                    { "Test.csproj", $"""
+                        <Project Sdk="Microsoft.NET.Sdk">
+                            <UsingTask TaskName="Microsoft.CodeAnalysis.BuildTasks.Csc" AssemblyFile="{_buildTaskDll}" />
+                            <PropertyGroup>
+                                <TargetFramework>netstandard2.0</TargetFramework>
+                                <UseSharedCompilation>{useSharedCompilation}</UseSharedCompilation>
+                            </PropertyGroup>
+                        </Project>
+                        """ },
+                });
+        }
+        finally
+        {
+            if (!useAppHost)
             {
-                { "File.cs", """
-                    class Program { static void Main() { System.Console.WriteLine("Hello from file"); } }
-                    """ },
-                { "Test.csproj", $"""
-                    <Project Sdk="Microsoft.NET.Sdk">
-                        <UsingTask TaskName="Microsoft.CodeAnalysis.BuildTasks.Csc" AssemblyFile="{_buildTaskDll}" />
-                        <PropertyGroup>
-                            <TargetFramework>netstandard2.0</TargetFramework>
-                            <UseSharedCompilation>{useSharedCompilation}</UseSharedCompilation>
-                        </PropertyGroup>
-                    </Project>
-                    """ },
-            });
+                File.Move(backupAppHost, originalAppHost);
+            }
+        }
 
         if (result == null) return;
 
         _output.WriteLine(result.Output);
 
         Assert.Equal(0, result.ExitCode);
         Assert.Contains(useSharedCompilation ? "server processed compilation" : "using command line tool by design", result.Output);
-        Assert.DoesNotContain("csc.dll", result.Output);
-        Assert.Contains(ExecutionConditionUtil.IsWindows ? "csc.exe" : "csc", result.Output);
+
+        if (useAppHost)
+        {
+            Assert.DoesNotContain("csc.dll", result.Output);
+            Assert.Contains($"csc{PlatformInformation.ExeExtension}", result.Output);
+        }
+        else
+        {
+            Assert.Contains("csc.dll", result.Output);
+            Assert.DoesNotContain("csc.exe", result.Output);
```

Comment:
Should this be `ExeExtension`? Or do we need a regex to say that `csc` isn't matched, start/end word boundaries around it?